src/scorecard.js

```diff
@@ -0,0 +1,73 @@
+var Scorecard = function(content) {
+  this.frameRecord = [];
+  this.totals = [];
+  this.DEFAULT_FRAMES = 10;
+  this.currentTurn = 0;
+  this.maxTurns = 20;
+  for (var i = 0; i < this.DEFAULT_FRAMES; i++) {
+    this.frameRecord.push(new content);
+  };
+};
+
+Scorecard.prototype.updateTotal = function() {
+  if (this.frameRecord[this.getCurrentFrame()].strike === true || this.frameRecord[this.getCurrentFrame()].spare === true) {
+    this.totals.push(10);
+  } else {
+    this.totals.push(this.frameRecord[this.getCurrentFrame()].result);  
+  };
+};
+
+Scorecard.prototype.checkLastFrameForStrike = function() {
+  if (this.getCurrentFrame() > 0 && this.frameRecord[this.getCurrentFrame() - 1].strike === true) return true;
+};
+
+Scorecard.prototype.checkLastFrameForSpare = function() {
+  if (this.getCurrentFrame() > 0 && this.frameRecord[this.getCurrentFrame() - 1].spare === true && this.frameRecord[this.getCurrentFrame()].firstBowl === null) return true;
+}
+
+Scorecard.prototype.checkTwoLastFramesForStrike = function() {
+  if (this.getCurrentFrame() > 1 && this.frameRecord[this.getCurrentFrame() - 2].strike === true && this.frameRecord[this.getCurrentFrame()].firstBowl === null) return true;
+}
+
+Scorecard.prototype.checkCurrentFrameForStrikeWith = function(pinsKnockedOver) {
+  if (this.checkEven(this.currentTurn) === true && pinsKnockedOver === 10) this.currentTurn +=1;
+}
+
+Scorecard.prototype.checkIfFrameIsOver = function() {
+  if (this.checkEven(this.currentTurn) != true) this.updateTotal();
+}
+
+Scorecard.prototype.bowl = function(pinsKnockedOver) {
```

Comment:
I'm not even going to try to read this method and neither will anyone else.  It is brittle, has multiple responsibilities, makes multiple changes to the state of the object; and worst of all, none of which is tested.
Sorry that's harsh - but I want to get across how important that would be in the real world :wink: